supervisor/jobs/__init__.py

```diff
@@ -73,10 +73,15 @@ class SupervisorJobError:
 
     type_: type[HassioError] = HassioError
     message: str = "Unknown error, see supervisor logs"
+    stage: str | None = None
 
     def as_dict(self) -> dict[str, str]:
         """Return dictionary representation."""
-        return {"type": self.type_.__name__, "message": self.message}
+        return {
+            "type": self.type_.__name__,
+            "message": self.message,
+            "stage": self.stage,
+        }
```

Comment:
This bends the job system to a specific requirement by the Backup system (so far we only have stages for backups). I wonder if it wouldn't be cleaner to just handle and return error "in-line" in Backup instead of relying on the Job system :thinking: 

But then, we already leaned in a lot into the Job system, and it seems to work. So :shrug: